### PR TITLE
feat(skills): add writing-unit-tests and writing-integration-tests skills

### PR DESCRIPTION
**Pull request by TheRealAgentK**
## Summary

Adds two new agent skills for writing pytest tests for Autohive integrations, plus updates to the existing upgrade skill.

### `writing-unit-tests` (mock_context + FetchResponse pattern)

- **File structure** — naming convention (`test_*_unit.py`), single vs split files for large integrations, boilerplate header
- **6 test categories** with concrete examples: happy path, request verification, error paths, edge cases, response shape, business logic
- **Mocking patterns** — single fetch, multi-fetch `side_effect`, exceptions, DELETE 204, `asyncio.sleep` patching, env vars
- **Coverage expectations** — target tests-per-action by integration size
- **7 common gotchas** — side_effect ordering, SDK validation, ActionError vs ActionResult, etc.

### `writing-integration-tests` (live_context + aiohttp pattern)

- **File structure** — naming convention (`test_*_integration.py`), double exclusion from CI (file naming + marker)
- **3 live_context fixture variants** — no auth (public APIs), API key auth, platform OAuth (token injection)
- **Environment variable handling** — token setup, `require_*` skip helpers, `.env.example` documentation
- **Destructive marker** — `@pytest.mark.destructive` for write operations, separate run commands
- **Test patterns** — chained tests (list → get), lifecycle tests (CRUD workflows), structural assertions
- **Coverage expectations** — target tests by integration size
- **Reference implementations** — hackernews, perplexity, bitly, hubspot

### `upgrading-sdk-v2` updates (Ninos's feedback)

- Added `ActionError` with `cost_usd` example for cost-bearing error paths (e.g. billable API call fails)
- Added conversion pattern for preserving `cost_usd` during migration
- Added step to remove `"error"` property from action output schemas in `config.json` after migrating to `ActionError`
- Added output schema cleanup to the checklist

## Context

Built while migrating the HubSpot integration (48 actions, 256 unit tests, 41 integration tests). Codifies the patterns that emerged from that work so future integrations get consistent test coverage.

### CODE REVIEW

File: skills/writing-unit-tests/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    return ctx
+```
+
+If the integration reads credentials from `context.auth`, populate it:
+
+```python
+ctx.auth = {
+    "auth_type": "PlatformOauth2",
+    "credentials": {"access_token": "test_token"},  # nosec B105
+}
```
[the remark this answers is not in the document]
Comment (TheRealAgentK) [reply skills/writing-unit-tests/SKILL.md:97]: Fair comment, addressed.

**Review by NinosMan — APPROVED** [review]
